stan/math/opencl/kernel_generator/unary_function_cl.hpp

```diff
@@ -0,0 +1,152 @@
+#ifndef STAN_MATH_OPENCL_KERNEL_GENERATOR_UNARY_FUNCTION_CL_HPP
+#define STAN_MATH_OPENCL_KERNEL_GENERATOR_UNARY_FUNCTION_CL_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/opencl/matrix_cl_view.hpp>
+#include <stan/math/opencl/err/check_matching_dims.hpp>
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/opencl/kernel_generator/type_str.hpp>
+#include <stan/math/opencl/kernel_generator/name_generator.hpp>
+#include <stan/math/opencl/kernel_generator/operation_cl.hpp>
+#include <stan/math/opencl/kernel_generator/as_operation_cl.hpp>
+#include <stan/math/opencl/kernel_generator/is_valid_expression.hpp>
+#include <string>
+#include <type_traits>
+#include <set>
+#include <utility>
+
+namespace stan {
+namespace math {
+
+/**
+ * Represents a unary function in kernel generator expressions.
+ * @tparam Derived derived type
+ * @tparam T type of argument
+ */
+template <typename Derived, typename T>
+class unary_function_cl
+    : public operation_cl<
+          Derived, typename std::remove_reference_t<T>::ReturnScalar, T> {
+ public:
+  using ReturnScalar = typename std::remove_reference_t<T>::ReturnScalar;
```

Comment:
Whups, just realized these won't work with value_type and scalar_type will they? I think you can just add a `Scalar` and `value_type` alias here and in binary_op and that should be fine

Reply:
I checked the code and both `value_type` and `scalar_type` will need specializations for kernel generator expressions. For now I am renaming ReturnScalar to Scalar. Changes to metaprogramming can come as a separate PR.